src/include/daos/debug.h

```diff
@@ -82,6 +82,12 @@ DAOS_FOREACH_LOG_FAC(D_LOG_DECLARE_FAC, DAOS_FOREACH_DB);
 
 /** initialize the debug system */
 int  daos_debug_init(char *logfile);
+/**
+ * DAOS-10412
+ * need this unnecessary internal API since Go can't see log masks due to
+ * no C pre-processor macro support
+ */
+int  daos_debug_init_ex(char *logfile, d_dbug_t logmask);
```

Comment:
Was the original goal to modify `daos_debug_init()` so that it has a new `logmask` parameter? I assume based on this comment that you wanted to be able to do `C.daos_debug_init(nil, C.DLOG_ERR)` in `src/control/cmd/daos/util.go`.

It's true that you can't use preprocessor macros directly from cgo, but there is a workaround. If you create a Go alias to the macro, the compiler takes care of it for you. So if you just apply the following patch to your PR, you can use the  Go aliases in `cmd/daos` and achieve your goal without making a new API.

```
--- a/src/control/lib/daos/constants.go
+++ b/src/control/lib/daos/constants.go
@@ -8,7 +8,26 @@ package daos

 import "time"

+/*
+#cgo LDFLAGS: -lgurt
+
+#include <gurt/dlog.h>
+*/
+import "C"
+
 const (
        // DefaultCartTimeout defines the default timeout for cart operations.
        DefaultCartTimeout = 60 * time.Second // Should use CRT_DEFAULT_TIMEOUT_S but it's not exported
 )
+
+const (
+       LogMaskEmit  = C.DLOG_EMIT
+       LogMaskEmerg = C.DLOG_EMERG
+       LogMaskAlert = C.DLOG_ALERT
+       LogMaskCrit  = C.DLOG_CRIT
+       LogMaskErr   = C.DLOG_ERR
+       LogMaskWarn  = C.DLOG_WARN
+       LogMaskNote  = C.DLOG_NOTE
+       LogMaskInfo  = C.DLOG_INFO
+       LogMaskDebug = C.DLOG_DBG
+)
```

With this patch in place, you could call `C.daos_debug_init(nil, daos.LogMaskErr)` and it would work fine (I just tested it here).

From my perspective, I don't really care either way. If you guys are happy with the new API, then you don't need to use my suggestion. I just wanted to point out that there is a way to do what I think you wanted to do.

Reply:
> Was the original goal to modify daos_debug_init() so that it has a new logmask parameter?

Yes, you are correct.

> From my perspective, I don't really care either way. If you guys are happy with the new API, then you don't need to use my suggestion. I just wanted to point out that there is a way to do what I think you wanted to do.

I don't really care too, I just wonder if doing so will not introduce a second location where log masks should be maintained ?
@jolivier23 , what do you think ??

Reply:
I'm fine either way.  My goal in filing the ticket was only to make the engine default INFO and this patch does that.

This other approach does have the minor downside of maintaining two lists as Bruno points out, but those masks are unlikely to change.